### PR TITLE
Add better esptool support (ArduinoIDE 2.0.0)

### PR DESCRIPTION
**Pull request by me-no-dev**
Fixes some compatibility issues with Arduino IDE 2.0.0

### CODE REVIEW

File: platform.txt [hunk 1]
```diff
@@ -187,12 +187,53 @@ recipe.size.pattern="{compiler.path}{compiler.size.cmd}" -A "{build.path}/{build
...
 tools.esptool_py.upload.protocol=esp32
 tools.esptool_py.upload.params.verbose=
 tools.esptool_py.upload.params.quiet=
-tools.esptool_py.upload.pattern_args=--chip {build.mcu} --port "{serial.port}" --baud {upload.speed} {upload.flags} --before default_reset --after hard_reset write_flash -z --flash_mode {build.flash_mode} --flash_freq {build.flash_freq} --flash_size {build.flash_size} 0xe000 "{runtime.platform.path}/tools/partitions/boot_app0.bin" {build.bootloader_addr} "{build.path}/{build.project_name}.bootloader.bin" 0x10000 "{build.path}/{build.project_name}.bin" 0x8000 "{build.path}/{build.project_name}.partitions.bin" {upload.extra_flags}
+tools.esptool_py.upload.pattern_args=--chip {build.mcu} --port "{serial.port}" --baud {upload.speed} {upload.flags} --before default_reset --after hard_reset write_flash -z --flash_mode {build.flash_mode} --flash_freq {build.flash_freq} --flash_size {build.flash_size} {build.bootloader_addr} "{build.path}/{build.project_name}.bootloader.bin" 0x8000 "{build.path}/{build.project_name}.partitions.bin" 0xe000 "{runtime.platform.path}/tools/partitions/boot_app0.bin" 0x10000 "{build.path}/{build.project_name}.bin"{upload.extra_flags}
```
[the remark this answers is not in the document]
Comment (me-no-dev) [reply platform.txt:209]: definitely not, but since none of the boards I use define it... it slipped through. nice catch.

**Comment by me-no-dev** [comment]
looks like I need to update the definitions for all boards, because they will not flash without `upload.tool.serial=esptool_py`. Have not tried OTA on the new IDE yet

**Comment by atanisoft** [comment]
> looks like I need to update the definitions for all boards

might be time to finally convert the boards.txt to a template generated file.

**Comment by me-no-dev** [comment]
> > looks like I need to update the definitions for all boards
> 
> might be time to finally convert the boards.txt to a template generated file.

it's on the list with gazillion tasks to do :D the new IDE is weird (or maybe it's the cli that is weird)

**Comment by atanisoft** [comment]
> the new IDE is weird (or maybe it's the cli that is weird)

Yeah, it's a complete rework and I think it is based on Eclipse now? Sadly they didn't migrate away from the rather fragile text file based configuration to something a bit more robust.

**Comment by me-no-dev** [comment]
it's based on Theia/Electron and written in PostScript. I dug a bit deeper into it, there are things broken and undocumented. They could have used the same text based system, but turn it into something like Make or CMake. Their build system is the same and is still slow.

**Pull request merged**